src/6model/serialization.c

```diff
@@ -177,8 +177,10 @@ static void * base64_decode(const char *s, size_t *data_len)
          || n[3] == -2
          || n[0] == -1
          || n[1] == -1
-         || (n[2] == -1 && n[3] != -1))
+         || (n[2] == -1 && n[3] != -1)) {
+            MVM_free(q);
```

Comment:
Actually the `q += 3` would never be reached if the long `if` expression is executed, since the function returns if the expression is true, thus the memory for `q` would still be leaked if `q` isn't freed at this point.  However, you're right, `MVM_free(data)` is necessary.  Actually, since `q` is assigned from data, shouldn't the correct code be this?:

    MVM_free(q);
    MVM_free(data);
    return NULL;